### PR TITLE
[SMALLFIX] Abstract out strategy class InodePermissionChecker.

### PR DESCRIPTION
**Pull request by uronce-cc**
This PR makes it easier to add extensions to the permission checker algorithm.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/InodePermissionChecker.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   *
+   * @param user the user
+   * @param groups the groups that user belongs to
+   * @param path the path
+   * @param inode inode of the path
+   * @param permission the permissions to check
+   * @throws AccessControlException when user does not have the permission
+   */
+  void checkPermission(String user, List<String> groups, String path, Inode<?> inode,
+      Mode.Bits permission) throws AccessControlException;
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/InodePermissionChecker.java]: Does `Mode.Bits` need to be extensible/generalized too?

File: core/server/master/src/main/java/alluxio/master/file/InodePermissionCheckerFactory.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package alluxio.master.file;
+
+/**
+ * Factory for {@link InodePermissionChecker}.
+ */
+public final class InodePermissionCheckerFactory {
+  /**
+   * @return a new instance of {@link InodePermissionChecker}
+   */
+  public static InodePermissionChecker create() {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/InodePermissionCheckerFactory.java]: I think with java 8, you can have a static create method implementation in an interface, so this functionality could probably go into the `InodePermissionChecker` interface definition.

File: core/server/master/src/main/java/alluxio/master/file/InodePermissionChecker.java (hunk 1 above)
Comment (uronce-cc) [reply core/server/master/src/main/java/alluxio/master/file/InodePermissionChecker.java]: In a future PR, `Mode.Bits` will be replaced by `Actions` which can internally be a `Set<Action>`, `Action` can be `READ`, `WRITE`, `EXECUTE`, `PIN`, `FREE`, etc.

For now, to keep the changes minimum, I'll keep the usage of Mode.Bits.

File: core/server/master/src/main/java/alluxio/master/file/InodePermissionCheckerFactory.java (hunk 2 above)
Comment (uronce-cc) [reply core/server/master/src/main/java/alluxio/master/file/InodePermissionCheckerFactory.java]: done

**Comment by uronce-cc** [comment]
jenkins, test this please

**Pull request merged**